components/places/src/bookmark_sync/store.rs

```diff
@@ -573,6 +625,7 @@ impl<'a> Store for BookmarksStore<'a> {
         &self,
         inbound: IncomingChangeset,
         incoming_telemetry: &mut telemetry::EngineIncoming,
+        validation: &mut Option<telemetry::Validation>,
```

Comment:
validation is a field on the `Engine`, so I wonder if we could change the `EngineIncoming` param to the `Engine`, to avoid the new param?